hnn_core/optimization.py

```diff
@@ -447,14 +583,17 @@ def _myoptrun(new_params):
         # tiny negative weights are possible. Clip them to 0.
         opt_results[opt_results < 0] = 0
```

Comment:
@rythorpe This is the part that seems redundant, the negative weights are trimmed inside `_opt_run()` as well as here

Reply:
Maybe the more realistic solution is to use a different optimization library that enforces hard boundaries. WDYT?